src/hyperion/external_interaction/ispyb/store_datacollection_in_ispyb.py

```diff
@@ -173,22 +178,34 @@ def _store_data_collection_group_table(self, conn: Connector) -> int:
         params["sampleid"] = self.ispyb_params.sample_id
         params["sample_barcode"] = self.ispyb_params.sample_barcode
 
-        return mx_acquisition.upsert_data_collection_group(list(params.values()))
+        return self._upsert_data_collection_group(conn, params)
+
+    @staticmethod
+    def _upsert_data_collection_group(
+        conn: Connector, params: StrictOrderedDict
+    ) -> int:
+        return conn.mx_acquisition.upsert_data_collection_group(list(params.values()))
+
+    @staticmethod
+    def _upsert_data_collection(conn: Connector, params: StrictOrderedDict) -> int:
```

Comment:
What's the difference between these two functions? Could we combine them?

Reply:
DataCollection and DataCollectionGroup are two different tables in ispyb - we probably want to keep them separate as sometimes we are doing an update and sometimes we are doing an insert, depending on whether we supply the row id.

There are a few bugs I have fixed in my workspace where we currently aren't supplying these on the `update_deposition()`